roles/postgresql/tasks/main.yml

```diff
@@ -0,0 +1,77 @@
+---
+- name: Pull PostgreSQL container image
+  containers.podman.podman_image:
+    name: "{{ postgresql_image }}"
+    state: present
+
+- name: Create PostgreSQL storage directory
+  ansible.builtin.file:
+    path: "{{ postgresql_data_dir }}"
+    state: directory
+    mode: "0700"
+    owner: 26
+    group: 26
+
+- name: Configure host based authentication
+  ansible.builtin.template:
+    src: pg_hba.conf.j2
+    dest: "{{ postgresql_data_dir }}/pg_hba.conf"
+    mode: "0644"
+
+- name: Create Podman secret for PostgreSQL admin password
+  containers.podman.podman_secret:
+    name: postgresql_admin_password
+    data: "{{ postgresql_admin_password }}"
+
+- name: Deploy PostgreSQL container
+  containers.podman.podman_container:
+    name: "{{ postgresql_container_name }}"
+    image: "{{ postgresql_image }}"
+    state: quadlet
+    healthcheck: pg_isready
+    sdnotify: healthy
+    network: host
+    volumes:
+      - "{{ postgresql_data_dir }}:/var/lib/pgsql/data:Z"
+      - "{{ postgresql_data_dir }}/pg_hba.conf:/var/lib/pgsql/data/pg_hba.conf:Z"
+    env:
+      POSTGRESQL_USER: "postgres"
+      POSTGRESQL_PASSWORD: "{{ postgresql_admin_password }}"
+      POSTGRESQL_DATABASE: "{{ postgresql_databases[0].name | default('foreman') }}"
+      POSTGRESQL_ADMIN_PASSWORD: "{{ postgresql_admin_password }}"
+    quadlet_options:
+      - |
+        [Install]
+        WantedBy=default.target
+
+- name: Create PostgreSQL users
+  community.postgresql.postgresql_user:
+    name: "{{ item.name }}"
+    password: "{{ item.password }}"
+    login_user: postgres
+    login_password: "{{ postgresql_admin_password }}"
+    login_host: localhost
+    state: present
+  loop: "{{ postgresql_users }}"
+  when: postgresql_users | length > 0
```

Comment:
You can omit the `when` here, if the array is empty, the loop will not be executed :)